cpp/src/arrow/compute/kernels/scalar_string.cc

```diff
@@ -2755,6 +2748,138 @@ Result<ValueDescr> StrptimeResolve(KernelContext* ctx, const std::vector<ValueDe
   return Status::Invalid("strptime does not provide default StrptimeOptions");
 }
 
+// ----------------------------------------------------------------------
+// string padding
+
+template <bool PadLeft, bool PadRight>
```

Comment:
I think these template parameters have little impact on performance.
Can we pass them as normal parameters?

Reply:
It's not really for performance, just that it's most convenient to pass them this way. Otherwise we'd need to create a subclass of StringTransformExecWithState for each kernel to pass the parameters, which would end up generating the same amount of code anyways. Or we could make them actual function options?

Reply:
The string pad kernels probably don't generate much code, so we can keep the template parameters IMHO.